.editorconfig

```diff
@@ -210,4 +210,9 @@ csharp_style_prefer_method_group_conversion = true:silent
 dotnet_style_operator_placement_when_wrapping = beginning_of_line
 tab_width = 4
 indent_size = 4
-end_of_line = crlf
\ No newline at end of file
+end_of_line = crlf
+
+# StyleCop rules
+dotnet_diagnostic.SA1400.severity = none  # Access modifier must be declared
+dotnet_diagnostic.SA1402.severity = none  # File may only contain a single type
+dotnet_diagnostic.SA1649.severity = none  # File name must match type name
```

Comment:
Should we put the file header rule in .editorconfig too? I believe VS knows to automatically add the header for new files when you do this, which is really nice when accepting PRs from external contributors.

Reply:
We have duplicate file header enforcement rules. IDE0073 and SA1633. I have set IDE0073 as a suggestion for now (and .editorconfig already had `file_header-template` configured). I am leaving in SA1633 as a warning while projects get onboarded one by one. Eventually though I will switch IDE0073 to warning, enable enforce style on build, and then turn off SA1633.